src/main/scala/firrtl/transforms/SimpleArithmetic.scala

```diff
@@ -0,0 +1,201 @@
+      case DoPrim(op, args, _, _) if isSimpleArithmeticExpr(op) => replaceSimple(expr)
+      case other => other
+    }
+  }
+
+  /** Inline bits in a Statement
+    *
+    * @param netlist a '''mutable''' HashMap mapping references to [[firrtl.ir.DefNode DefNode]]s to their connected
+     * [[firrtl.ir.Expression Expression]]s. This function '''will''' mutate it if stmt is a [[firrtl.ir.DefNode
+    * DefNode]] with a Temporary name and a value that is a [[PrimOp]] Bits
+    * @param stmt the Statement being searched for nodes and transformed
+    * @return Returns stmt with Bits inlined
+    */
+  def onStmt(netlist: Netlist, symbolTable: Map[WrappedExpression, Int])(stmt: Statement): Statement = {
+    println(s"stmt = ${stmt.serialize}")
+    stmt.map(onStmt(netlist, symbolTable)).map(onExpr(netlist, symbolTable))
+  }
+
+  /** Replaces bits in a Module */
+  def onMod(mod: DefModule,
+    netlist: Netlist,
+    symbolTable: Map[WrappedExpression, Int]): DefModule =
+    mod.map(onStmt(netlist, symbolTable))
+}
+
+/**
+  * {{{
+  *   circuit Top :
+  *   module Top :
+  *     input a : SInt<2>
+  *     output y : SInt<2>
+  *     y <= add(a, SInt(-1))
+  * }}}
+  *
+  * converts to
+  *
+  * {{{
+  *   circuit Top :
+  *   module Top :
+  *     input a : SInt<2>
+  *     output y : SInt<2>
+  *     y <= sub(a, SInt(1))
+  * }}}
+  *
+  * ________________________
+  *
+  * * {{{
+  *   *   circuit Top :
+  *   *   module Top :
+  *   *     input a : SInt<2>
+  *   *     output y : SInt<2>
+  *   *     y <= sub(a, SInt(-1))
+  *   * }}}
+  * *
+  * * converts to
+  * *
+  * * {{{
+  *   *   circuit Top :
+  *   *   module Top :
+  *   *     input a : SInt<2>
+  *   *     output y : SInt<2>
+  *   *     y <= add(a, SInt(1))
+  *   * }}}
+   */
+class SimpleArithmetic extends Transform {
+  def inputForm = LowForm
+  def outputForm = LowForm
+
+  import SimpleArithmetic._
+
+  def makeModuleInfo(c: Circuit): (Map[String, Map[String, String]], Seq[String], InstanceGraph, Map[String, DefModule]) = {
+    val iGraph = (new InstanceGraph(c))
+    val childInstances = iGraph.getChildrenInstances.mapValues(_.map(inst => inst.name -> inst.module).toMap).toMap
+
+    // Order from leaf modules to root so that any module driving an output
+    // with a constant will be visible to modules that instantiate it
+    // TODO Generating order as we execute constant propagation on each module would be faster
+    val x = iGraph.moduleOrder.reverse
+
+    val moduleNameTraversalOrder = x.map{ m => m.name }
+
+    val modNameModules = x.map{ m => (m.name -> m)}.toMap
+    (childInstances, moduleNameTraversalOrder, iGraph, modNameModules)
+  }
+
+  /** Build a [[Netlist]] from a Module's connections and Nodes
+    * This assumes [[firrtl.LowForm LowForm]]
+    * @param mod [[firrtl.ir.Module Module]] from which to build a [[Netlist]]
+    * @return [[Netlist]] of the module's connections and nodes
+    */
+  def buildNetlist(mod: DefModule): Netlist = {
```

Comment:
Perhaps we should wait until after we merge your other PR, so that we can reuse this function?